client/asset/dcr/dcr.go

```diff
@@ -1640,9 +1666,8 @@ func (dcr *ExchangeWallet) FindRedemption(ctx context.Context, coinID dex.Bytes)
 	if contractBlock == nil {
 		dcr.findRedemptionsInMempool([]outPoint{contractOutpoint})
 	} else {
-		dcr.tipMtx.RLock()
-		bestBlock := dcr.currentTip
-		dcr.tipMtx.RUnlock()
+		bestHash, bestHeight := dcr.blockCache.Tip()
```

Comment:
Looking to get this PR in now that the 0.3 cycle is well underway.  When it's rebased, the new method `calcPastMedianTime()` needs to use `bestHash, _ := dcr.blockCache.Tip()` (and a change to the test) but other than that, it looks good.
Thanks @itswisdomagain!